sdk/burdock/reader/sql/databricks.py

```diff
@@ -0,0 +1,97 @@
+import datetime
+from pyspark.sql import Row
+from pyspark.sql.types import BooleanType
+from burdock.sql import CollectionMetadata
+from burdock.sql import PrivateReader
+from burdock.mechanisms.gaussian import Gaussian
+
+class DatabricksSparkReader:
+    def __init__(self, sparkSession, meta_path):
+        self.spark = sparkSession
+        self.schema = CollectionMetadata.from_file(meta_path)
+        self.private_reader = PrivateReader(None, self.schema, 1.0)
+
+    """
+        Executes a raw SQL string against the Spark Hive database and returns spark dataframe.
+    """
+    def execute(self, query):
+        if not isinstance(query, str):
+            raise ValueError("Please pass strings to execute.  To execute ASTs, use execute_typed.")        
+        
+        max_contrib = self.private_reader.max_contrib
+        epsilon = self.private_reader.epsilon
+        interval_widths = self.private_reader.interval_widths
+
+        subquery, fullquery = self.private_reader.rewrite(query)
+        syms = subquery.all_symbols()
+        numeric_cols = subquery.numeric_symbols()
+
+        if numeric_cols is None:
+          pass
+
+        subquery_df = self.spark.sql(str(subquery))
+        noisedDF = subquery_df.rdd.mapPartitions(applyNoises(numeric_cols, epsilon, max_contrib, interval_widths)).toDF()
+        
+        # censor dimensions for privacy
+        if subquery.agg is not None:
+            noisedDF = noisedDF.filter(noisedDF.keycount > max_contrib ** 2)
+
+        noisedDF.createOrReplaceTempView("query_df_noised")
+        noisedDF.cache()          
+
+        q_result = self.spark.sql(str(fullquery))
+        return q_result

```

Comment:
(2) fullquery
I need some help to change fullquery for databricks spark.
basically replace "subquery" with a temp table string.
for example, for PUMs example, it can be 
"SELECT ( SUM ( count_0xe3f0 ) ) AS c, married AS m FROM **query_df_noised** AS exact_aggregates GROUP BY married ORDER BY c, m DESC"
replace the subquery with "query_df_noised"